### ISSUE TITLE
Querying over entity objects with multiple bidirectional association objects caused stackoverflow

### ISSUE THREAD

**Issue by chhavigangwal** [Impetus/kundera#691]

**Comment by pragalbhgarg** [comment]
To further demonstrate this issue, lets say we have 3 entity classes namely Books, Categories and Languages.
1. Books and Categories are bidirectionally associated i.e. Many Books can belong to a single Category and One Category can have many Books. The fetch type is 'EAGER' both side.
2. Similarly, Books and Languages are bidirectionally associated with fetch type being 'EAGER' both side.

Books.java

```
@Entity
@Table(name = "books")
public class Books {
    @Id
    @Column(name = "book_id")
    private int bookId;

    private String title;

    private String author;

    @ManyToOne(cascade=CascadeType.ALL,fetch = FetchType.EAGER)
    @JoinColumn(name = "language")
    private Languages language;

    @ManyToOne(cascade=CascadeType.ALL,fetch = FetchType.EAGER)
    @JoinColumn(name = "category")
    private Categories category;

   //setter and getters
}
```

Categories.java

```
@Entity
@Table(name = "book_categories")
public class Categories {

    @Id
    @Column(name = "category_id")
    private int categoryId;

    @Column(name = "category_name")
    private String categoryName;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "category")
    private List<Books> books;

    //setters and getters
}
```

Languages.java

```
@Entity
@Table(name = "book_languages")
public class Languages {

    @Id
    @Column(name = "language_id")
    private int languageId;

    private String language;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "language")
    private List<Books> books;   

    //setters and getters
}
```

Now, while querying over any of these entity classes, the code throws a stack overflow exception. The exception is mainly due to eager fetching which causes the code to enter an infinite recursive loop while fetching associated objects.

**Comment by chhavigangwal** [comment]
Fixed. Released with Kundera-2.15.

Chhavi
